### PR TITLE
Add additionalMetadataNavigation Page

### PR DESCRIPTION
**Pull request by thanhz**

### CODE REVIEW

**Review by TomJKing — COMMENTED** [review]
Looks good. Just some minor comments and suggestions.

Also I've proposed a slight refactoring of the updating of the cache, here: https://github.com/nationalarchives/tdr-transfer-frontend/tree/TDR-2089-pull-request-feedback-cache 

This is because it would be good to try contain all the cache updating in a single function, to help ensure operations don't contradict each other and mess up what is stored in the cache.

File: test/controllers/AdditionalMetadataNavigationControllerSpec.scala [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+      val response = controller.getPaginatedFiles(consignmentId, page, limit = None, selectedFolderId = selectedFolderId)
+        .apply(FakeRequest(GET, s"/consignment/$consignmentId/additional-metadata/$selectedFolderId/$page").withCSRFToken)
+
+      status(response) mustBe FORBIDDEN
+    }
+
+    "will return forbidden if the user does not own the consignment on the file selection page" in {
+      val selectedFolderId = UUID.randomUUID()
+      val page = 1
+      setConsignmentTypeResponse(wiremockServer, "judgment")
```
[the remark this answers is not in the document]
Comment (thanhz) [reply test/controllers/AdditionalMetadataNavigationControllerSpec.scala]: Nope, I've corrected it

File: app/controllers/AdditionalMetadataNavigationController.scala [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+          navigationForm.fill(NodesFormData(nodesToDisplay, selected = List(), allNodes = List(), page, selectedFolderId.toString,
+            paginatedFiles.parentFolder)))
+        ).uncache()
+      }
+  }
+
+  def submit(consignmentId: UUID, limit: Option[Int], selectedFolderId: UUID): Action[AnyContent] = standardTypeAction(consignmentId) {
+    implicit request: Request[AnyContent] =>
+      val errorFunction: Form[NodesFormData] => Future[Result] = { formWithErrors: Form[NodesFormData] =>
+        Future(Ok)
```
[the remark this answers is not in the document]
Comment (thanhz) [reply app/controllers/AdditionalMetadataNavigationController.scala:94]: Agreed. I've just left it as it is now for when we know what the behavior should be

**Comment by thanhz** [comment]
> Looks good. Just some minor comments and suggestions.
> 
> Also I've proposed a slight refactoring of the updating of the cache, here: https://github.com/nationalarchives/tdr-transfer-frontend/tree/TDR-2089-pull-request-feedback-cache
> 
> This is because it would be good to try contain all the cache updating in a single function, to help ensure operations don't contradict each other and mess up what is stored in the cache.

That makes sense. I've had to change the adding and deselecting methods slightly as I was getting the error `ERR wrong number of arguments for 'srem' command`.

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks

**Pull request merged**